repos/iiif-storage/src/Utility/CheapOmekaRelationshipRequest.php

```diff
@@ -188,35 +197,90 @@ public function getManifestFromCanvas(int $canvasId)
         return $this->manifestsByCanvasId[$canvasId]; // keys: [ 'omeka_id', 'uri', 'label']
     }
 
-    public function manifestExists(string $manifestUri)
+    /**
+     * @param string $manifestUri
+     * @param string|null $baseUrl
+     * @return int|false
+     */
+    public function manifestExists(string $manifestUri, string $baseUrl = null)
     {
-        return $this->resourceExists('sc:Manifest', $manifestUri);
+        return $this->resourceExists('sc:Manifest', $manifestUri, $baseUrl);
     }
 
-    public function canvasExists(string $canvasUri)
+    /**
+     * @param string $canvasUri
+     * @param string|null $baseUrl
+     * @return int|false
+     */
+    public function canvasExists(string $canvasUri, string $baseUrl = null)
     {
-        return $this->resourceExists('sc:Canvas', $canvasUri);
+        return $this->resourceExists('sc:Canvas', $canvasUri, $baseUrl);
     }
 
-    public function collectionExists(string $collectionUri)
+    /**
+     * @param string $collectionUri
+     * @param string|null $baseUrl
+     * @return int|false
+     */
+    public function collectionExists(string $collectionUri, string $baseUrl = null)
     {
-        return $this->resourceExists('sc:Collection', $collectionUri);
+        return $this->resourceExists('sc:Collection', $collectionUri, $baseUrl);
     }
 
-    public function resourceExists(string $resource, string $id): bool
+    /**
+     * @param string $resource
+     * @param string $id
+     * @param string|null $baseUrl
+     * @return int | false
+     */
+    public function resourceExists(string $resource, string $id, string $baseUrl = null)
```

Comment:
Should callers just be invoking this instead of `collectionExists`, etc? If not, might be better to make this non-public.